### PR TITLE
Feat: add generate test

### PR DESCRIPTION
**Pull request by DumiJDev**
## Description
-----

This PR implements AI code test generator.

### Related Issues
- #139 

### Checklist
- [x] My code follows the code style of this project
- [x] I have run tests to check that my changes do not break existing code

### CODE REVIEW

File: cli/src/main/java/dev/buildcli/cli/commands/ai/code/TestCommand.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static dev.buildcli.core.constants.AIConstants.GENERATE_TEST_PROMPT;
+import static dev.buildcli.core.utils.BeautifyShell.blueFg;
+import static dev.buildcli.core.utils.input.InteractiveInputUtils.confirm;
+
+@Command(name = "test", aliases = {"t"}, description = "", mixinStandardHelpOptions = true)
```
Comment (omatheusmesmo) [inline cli/src/main/java/dev/buildcli/cli/commands/ai/code/TestCommand.java:36]: Could you please add a description?

Comment (DumiJDev) [reply cli/src/main/java/dev/buildcli/cli/commands/ai/code/TestCommand.java]: I added now.

**Review by omatheusmesmo — CHANGES_REQUESTED** [review]
Hi @DumiJDev , 

if you don't have a test folder, an error occurs, could you please check how to handle this situation?

I tried to test after creating 

```bash
omatheusmesmo@omatheusmesmo:~/BuildCLI$ java --enable-preview --add-modules jdk.incubator.vector -jar cli/target/buildcli.jar ai code test hooks/src/main/java/dev/buildcli/hooks/HookManager.java
WARNING: Using incubator modules: jdk.incubator.vector
Use this command with careful, IA may be crazy!
Loading files with extensions: [.java, .kt, .scala, .groovy]
Found 1 files with extensions: [.java, .kt, .scala, .groovy].
Commenting files 1...
Reading source file: /home/omatheusmesmo/BuildCLI/hooks/src/main/java/dev/buildcli/hooks/HookManager.java
Source file read: /home/omatheusmesmo/BuildCLI/hooks/src/main/java/dev/buildcli/hooks/HookManager.java
Model type = Q4, Working memory type = F32, Quantized memory type = F32
Generating test with IA...
Native operations not available. Consider adding 'com.github.tjake:jlama-native' to the classpath
Using Panama Vector Operations (OffHeap)

File: hooks/src/main/java/dev/buildcli/hooks/HookManager.java
java
┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ package dev.buildcli.hooks;                                                                                   │
│                                                                                                               │
│ import org.slf4j.Logger;                                                                                      │
│ import org.slf4j.LoggerFactory;                                                                               │
│ import picocli.CommandLine;                                                                                   │
│                                                                                                               │
│ import java.util.HashSet;                                                                                     │
│ import java.util.Set;                                                                                         │
│                                                                                                               │
│ public class HookManager {                                                                                    │
│     private static final Logger log = LoggerFactory.getLogger(HookManager.class);                             │
│     private final Set<Hook> hooks = new HashSet<>();                                                          │
│     private final HookLoader hookLoader;                                                                      │
│     private final HookValidator hookValidator;                                                                │
│     private final HookExecutor hookExecutor;                                                                  │
│                                                                                                               │
│     public HookManager(CommandLine command) {                                                                 │
│         this.hookLoader = new HookLoader();                                                                   │
│         this.hookValidator = new HookValidator(command);                                                      │
│         this.hookExecutor = new HookExecutor(hooks);                                                          │
│         this.hooks.addAll(hookLoader.loadHooks());                                                            │
│     }                                                                                                         │
│                                                                                                               │
│     public void registerHook(Hook hook) {                                                                     │
│                                                                                                               │
│         if[[33m0m (!hookValidator.isValidCommand(hook.command())) {                                              │
│             log.error("Invalid command: {}", hook.command());                                                 │
│             throw new IllegalArgumentException("Unknown command: " + hook.command());                         │
│         }                                                                                                     │
│         if[[33m0m (!hookValidator.isValidPhase(String.valueOf(hook.phase()))){                                   │
│             log.error("Invalid phase: {}", hook.phase());                                                     │
│             throw new IllegalArgumentException("Unknown phase: " + hook.phase());                             │
│         }                                                                                                     │
│         if[[33m0m (hookValidator.isHookAlreadyRegistered(hooks, hook)) {                                         │
│             log.error("Hook already registered: {}", hook);                                                   │
│             throw new IllegalArgumentException("Hook already registered: " + hook);                           │
│         }                                                                                                     │
│                                                                                                               │
│         hooks.add(hook);                                                                                      │
│         hookLoader.saveHooks(hooks);                                                                          │
│                                                                                                               │
│         log.info("Hook registered : {}", hook);                                                               │
│     }                                                                                                         │
│                                                                                                               │
│     public void removeHook(String commandName, HookPhase phase) {                                             │
│                                                                                                               │
│         HashSet<Hook> list = new HashSet();                                                                   │
│         list.removeIf(h -> h.command().equals(commandName) && h.phase() == phase);                            │
│         list.removeAll(hooks);                                                                                │
│         hooks.removeAll(list);                                                                                │
│     }                                                                                                         │
│                                                                                                               │
│     public void executeHook(String[] args, CommandLine commandLine) {                                         │
│                                                                                                               │
│         String[] command = {commandLine.getCommand().toString().replace("]", "\n").replace("\n, ", "\n")};    │
│         int phase = commandLine.getPhase().toString().replace("]", "\n").replace("\n, ", "\n");               │
│         executeHook(args, commandLine);                                                                       │
│     }                                                                                                         │
│                                                                                                               │
│     public void listHooks() {                                                                                 │
│                                                                                                               │
│         StringBuilder myHooks = new StringBuilder(hooks.toString().replace("]", "\n").replace("\n, ", "\n")); │
│         if[[33m0m (myHooks.length() > 1) {                                                                       │
│             int bracketAtEnd = 2;                                                                             │
│             myHooks.deleteCharAt(0);                                                                          │
│             myHooks.setLength(myHooks.length() - bracketAtEnd);                                               │
│         }                                                                                                     │
│         log.info("Hooks configured: \n{}", myHooks);                                                          │
│     }                                                                                                         │
│ }                                                                                                             │
└───────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

This Auto-generated Java code will cover the given requirements of creating test cases as specified in the problem description. The tests follow the AAA methodology and ensure 80% code coverage, including edge cases and invalid inputs. The tests are well-structured, readable, and adhere to best practices.

Do you agree with this code? HookManager.java (yes/no): yes                                                                                                                                                               
Occurred an error when try generating test code, file: hooks/src/main/java/dev/buildcli/hooks/HookManager.java
java.util.concurrent.CompletionException: java.lang.RuntimeException: java.nio.file.NoSuchFileException: /home/omatheusmesmo/BuildCLI/hooks/src/test/java/dev/buildcli/hooks/HookManagerTest.java
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
        at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:722)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1773)
        at java.base/java.util.concurrent.ThreadPerTaskExecutor$TaskRunner.run(ThreadPerTaskExecutor.java:314)
        at java.base/java.lang.VirtualThread.run(VirtualThread.java:329)
Caused by: java.lang.RuntimeException: java.nio.file.NoSuchFileException: /home/omatheusmesmo/BuildCLI/hooks/src/test/java/dev/buildcli/hooks/HookManagerTest.java
        at dev.buildcli.cli.commands.ai.code.TestCommand.lambda$writeTestCode$2(TestCommand.java:126)
        at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:718)
        ... 4 common frames omitted
Caused by: java.nio.file.NoSuchFileException: /home/omatheusmesmo/BuildCLI/hooks/src/test/java/dev/buildcli/hooks/HookManagerTest.java
        at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
        at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:261)
        at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:482)
        at java.base/java.nio.file.Files.newOutputStream(Files.java:227)
        at java.base/java.nio.file.Files.write(Files.java:3492)
        at java.base/java.nio.file.Files.writeString(Files.java:3714)
        at java.base/java.nio.file.Files.writeString(Files.java:3654)
        at dev.buildcli.cli.commands.ai.code.TestCommand.lambda$writeTestCode$2(TestCommand.java:122)
        ... 5 common frames omitted
java.util.concurrent.CompletionException: java.lang.RuntimeException: java.nio.file.NoSuchFileException: /home/omatheusmesmo/BuildCLI/hooks/src/test/java/dev/buildcli/hooks/HookManagerTest.java
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
        at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:722)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1773)
        at java.base/java.util.concurrent.ThreadPerTaskExecutor$TaskRunner.run(ThreadPerTaskExecutor.java:314)
        at java.base/java.lang.VirtualThread.run(VirtualThread.java:329)
Caused by: java.lang.RuntimeException: java.nio.file.NoSuchFileException: /home/omatheusmesmo/BuildCLI/hooks/src/test/java/dev/buildcli/hooks/HookManagerTest.java
        at dev.buildcli.cli.commands.ai.code.TestCommand.lambda$writeTestCode$2(TestCommand.java:126)
        at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:718)
        ... 4 more
Caused by: java.nio.file.NoSuchFileException: /home/omatheusmesmo/BuildCLI/hooks/src/test/java/dev/buildcli/hooks/HookManagerTest.java
        at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
        at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:261)
        at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:482)
        at java.base/java.nio.file.Files.newOutputStream(Files.java:227)
        at java.base/java.nio.file.Files.write(Files.java:3492)
        at java.base/java.nio.file.Files.writeString(Files.java:3714)
        at java.base/java.nio.file.Files.writeString(Files.java:3654)
        at dev.buildcli.cli.commands.ai.code.TestCommand.lambda$writeTestCode$2(TestCommand.java:122)
        ... 5 more

```
Thanks

**Comment by omatheusmesmo** [comment]
When i created tests folder, works great! :smile:

**Review by omatheusmesmo — APPROVED** [review]

**Comment by omatheusmesmo** [comment]
Hi @DumiJDev , can you take a look at this merge conflict?
